src/decoder_opus.cpp

```diff
@@ -24,26 +24,33 @@
 #include "audio_decoder.h"
 #include "decoder_opus.h"
 
-static int custom_read(void* stream, unsigned char* ptr, int nbytes) {
-	FILE* f = reinterpret_cast<FILE*>(stream);
-	return fread(ptr, 1, nbytes, f);
+static int vio_read_func(void* stream, unsigned char* ptr, int nbytes) {
+	auto* f = reinterpret_cast<Filesystem::InputStreamRaw*>(stream);
+	if (nbytes == 0) return 0;
+	return (int)(f->read(reinterpret_cast<char*>(ptr), nbytes).gcount());
 }
 
-static int custom_seek(void* stream, opus_int64 offset, int whence) {
-	FILE* f = reinterpret_cast<FILE*>(stream);
-	return fseek(f, offset, whence);
-}
+static int vio_seek_func(void* stream, opus_int64 offset, int whence) {
+	auto* f = reinterpret_cast<Filesystem::InputStreamRaw*>(stream);
+	if (f->eof()) f->clear(); //emulate behaviour of fseek
+
+	f->seekg(offset, Filesystem::CSeekdirToCppSeekdir(whence));
 
-static opus_int64 custom_tell(void* stream) {
-	FILE* f = reinterpret_cast<FILE*>(stream);
-	return ftell(f);
+	return 0;
 }
 
-static int custom_close(void* stream) {
-	FILE* f = reinterpret_cast<FILE*>(stream);
-	return fclose(f);
+static opus_int64 vio_tell_func(void* stream) {
+	auto* f = reinterpret_cast<Filesystem::InputStreamRaw*>(stream);
+	return f->tellg();
```

Comment:
I'm pretty sure this is `int64_t` everywhere, but maybe we want  a `static_cast<opus_int64>` just in case?